### PR TITLE
[IMP] payment(_*): normalize logs across all acquirers

### PR DESCRIPTION
**Pull request by hote-odoo**
The logs for payments contain the transaction reference whenever possible.
Before logs for transactions contained the reference or the id of the
transaction in an inconsistent way. Now transactions are identified by
reference whenever possible.

The logs for payments for the same function on different acquirers should
have the same format. Same flow steps for different acquirers had
information passed in different formats. Now at each step of a transaction
flow log messages have the same format regardless of the acquirer.

Overall the payment logs should have an uniform format. Hopefully
understanding log messages related to transactions should be easier, as
now log format is independent of the acquirer and transaction are easily
identified by reference.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/payment/models/payment_transaction.py
Comment (Feyensv) [inline addons/payment/models/payment_transaction.py:793]: tx_id is not used anymore and can be removed from logging values

**Review by Feyensv — COMMENTED** [review]
Honestly, I expected less diff, but it makes sense :D.

**Review by AntoineVDV — COMMENTED** [review]
Since this is your first task reaching the review stage I'll first elaborate a bit about:

The commit message:
- The basic title format is: `[TAG] module(s)_name(s): short description` (no dot at the end) where `module(s)_name(s)` usually consists of a single module but can also use parenthesis or expanding braces to cover more than one module. Change your commit title to `[IMP] payment(_*): normalize logs across all acquirers`.
- Both the commit message title and commit body should have a max line length of 72 chars.
- The commit message body should be used to comment on the *why* of the change. The description can be arbitrarily long. Here, you mainly describe what you changed and the rules that you followed to get to that change, but you don't explain *why* the code is better after your change.

The review process:
- Kudos for creating this PR as a draft and keeping it in this state until it's ready for review. You could even have flagged it as ready for review as soon as the task reached the Testing BE [green] stage so that eventual code owners can have a look.
- When you receive a review comment, try to take it into account quickly enough to avoid blocking the task. By "taking into account" I mean:
  - Apply *and* resolve the comment if you agree with it.
  - Explain why you disagree with it if you do, and don't resolve it.
- After comments have all been replied to or resolved, re-request a review from the reviewer.

**Comment by Feyensv** [comment]
> * When you receive a review comment, try to take it into account quickly enough to avoid blocking the task. By "taking into account" I mean:
>   
>   * Apply _and_ resolve the comment if you agree with it.
>   * Explain why you disagree with it if you do, and don't resolve it.

I said he could wait for your review before applying those comments, since they were non blocking and I was sure you would have sthg to say on the PR subject :D.

File: addons/payment/models/payment_transaction.py
Comment (AntoineVDV) [inline addons/payment/models/payment_transaction.py:786]: In our scope, we prefer wrapping long lines as follows when the function's parameters don't all fit on one line:
```suggestion
                _logger.warning(
                    "invalid callback signature for transaction with reference %s",
                    tx.reference
                )
```
However, they do fit in this case so you can simply do this:
```suggestion
                _logger.warning(
                    "invalid callback signature for transaction with reference %s", tx.reference
                )
```

File: addons/payment/models/payment_transaction.py
Comment (AntoineVDV) [inline addons/payment/models/payment_transaction.py:794]:
```suggestion
                    'reference': tx.reference,
```
or
```suggestion
                    'ref': tx.reference,
```

File: addons/payment/models/payment_transaction.py
Comment (AntoineVDV) [inline addons/payment/models/payment_transaction.py:844]: Note in case you'd get too carried away when applying the comments above: this one is fine. Even if the params are not named, it's clear enough and convenient that it fits on only two lines.

File: addons/payment/models/payment_transaction.py
Comment (AntoineVDV) [inline addons/payment/models/payment_transaction.py:885]: Same thing for this one: it's good as it is :+1:

File: addons/payment_adyen/models/payment_transaction.py
Comment (AntoineVDV) [inline addons/payment_adyen/models/payment_transaction.py:237]:
```suggestion
            _logger.warning("the transaction with reference %s was cancelled. reason: %s",
```

**Review by AntoineVDV — CHANGES_REQUESTED** [review]
I only reviewed one file because I believe that, after applying my comments to the entire diff, almost everything should be good to go. I'll make another pass when all the comments are resolved and applied where they make sense.

**Review by AntoineVDV — APPROVED** [review]
I applied some nitpicking changes in a fixup commit. I let your read through it and if the suggested changes are ok for you, squash the commits so we can merge the PR :)

File: addons/payment/controllers/post_processing.py
Comment (hote-odoo) [reply addons/payment/controllers/post_processing.py:89]: I was not used anywhere, the logging message uses directly the error e.

Comment (AntoineVDV) [reply addons/payment/controllers/post_processing.py:89]: It's returned to the JS a few lines below.

Comment (hote-odoo) [reply addons/payment/controllers/post_processing.py:89]: Indeed, thank you !

**Comment by AntoineVDV** [comment]
@robodoo r+

**Pull request closed without merging**
